packages/astro/test/lit-element.test.js

```diff
@@ -28,4 +28,13 @@ LitElement.skip('Renders a custom element by the constructor', async ({ runtime
   assert.ok($('my-element').html().includes(`<div>Testing...</div>`), 'shadow rendered');
 });
 
+// The Lit renderer adds browser globals that interfere with other tests, so remove them now.
```

Comment:
Is this going to be a problem for Astro? Will components which try to access `window` after the Lit renderer has run still get an error?

Reply:
It's a problem if you are using code that does something like `if(typeof window !== 'undefined') ...` which unfortunately there's still a lot of such code out there in the wild. But you are opting into this problem when you use the Lit renderer. Which means you might not be able to use it with some libraries and some other frameworks.

Reply:
Ah okay. This definitely has the potential to break a lot of assumptions made in library code. Can we make sure this is clearly documented in the renderer's readme?

Shame that the `experimental-vm` feature isn't an option for us ☹️

Reply:
There are much better ways to check which environment your code is running in (Deno has a `window`) object. The gold standard for Node testing is `typeof process === 'object' && Object.prototype.toString.call(process) === '[object process]';`

I wouldn't want to go the vm route anyways, as it would mean that code can't communicate with each other (you couldn't import a Lit element and use another of its exports). Shimming is easier to deal with actually.

Reply:
Added docs